biohansel/cli.py

```diff
@@ -0,0 +1,281 @@
+import logging
+from typing import Union, Optional, Tuple, List
+
+import attr
+import click
+import pandas as pd
+
+from biohansel.subtype import subtype_contigs_samples, subtype_reads_samples, Subtype
+from biohansel.subtype.const import SUBTYPE_SUMMARY_COLS, JSON_EXT_TMPL
+from biohansel.subtype.metadata import read_metadata_table, merge_metadata_with_summary_results
+from biohansel.subtype.subtype_stats import subtype_counts
+from biohansel.subtype.util import get_scheme_fasta, init_subtyping_params
+from biohansel.utils import does_file_exist, collect_inputs
+from biohansel.utils import init_console_logger
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+
+
+@click.group(context_settings=CONTEXT_SETTINGS)
+@click.version_option()
+@click.option('-v', '--verbose', count=True,
+              help="Logging verbosity (-v for logging warnings; -vvv for logging debug info)")
+def cli(verbose):
+    """Subtype with a biohansel scheme or create a scheme for your organism of interest
+    """
+    lvl = init_console_logger(verbose)
+    logging.debug('Initialized logging with %s level', lvl)
+
+
+def check_between_0_and_1_inclusive(ctx: click.Context,
+                                    param: click.Option,
+                                    value: Optional[Union[int, float]]) -> Optional[Union[int, float]]:
+    if value is None or 0.0 <= value <= 1.0:
+        return value
+    else:
+        raise click.BadParameter('value needs to be between 0.0 and 1.0 inclusive!')
+
+
+def check_positive_number(ctx: click.Context,
```

Comment:
same for here?

Reply:
See http://click.pocoo.org/6/options/#callbacks-for-validation

Reply:
ok yup that makes sense